providers/implementations/signature/rsa.c

```diff
@@ -336,9 +336,9 @@ static int rsa_signature_init(void *vprsactx, void *vrsa, int operation)
                     return 0;
                 }
 
-                strncpy(prsactx->mdname, mdname, sizeof(prsactx->mdname));
+                strncpy(prsactx->mdname, mdname, sizeof(prsactx->mdname) - 1);
                 strncpy(prsactx->mgf1_mdname, mgf1mdname,
-                        sizeof(prsactx->mgf1_mdname));
+                        sizeof(prsactx->mgf1_mdname) - 1);
                 prsactx->saltlen = min_saltlen;
```

Comment:
This seems to be a change that in practice doesn't actually fix anything. mdname seems to be OSSL_MAX_NAME_SIZE, 50, and the longest we're going to put into currently looks like it's going to be 12+1.

If you want to future proof this, I think we need to check that it actually fits in it, and that if it's too small generate an error. If we don't mind truncation, I suggest you just replace strncpy by OPENSSL_strlcpy() and keep the size.

Reply:
For these ones I am using `OPENSSL_strlcpy()` and raising an error if truncation happens.